src/cmd/gazebo.bash_completion.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+#
+# Copyright (C) 2022 Open Source Robotics Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# bash tab-completion
+
+# This is a per-library function definition, used in conjunction with the
+# top-level entry point in ign-tools.
+
+GZ_SIM_COMPLETION_LIST="
+  -g
+  --iterations
+  --levels
+  --headless-rendering
```

Comment:
@mabelzhang , I'm addressing the TODO to add `--headless-rendering` in this PR. A review would be appreciated

Reply:
Looks good, test is passing to me, and manual output of `--help` matches this.
I think this list tries to follow the order of flags in the help message so that it's easier for manual checks. In that case, `--headless-rendering` should go between `--record-path` and `--record-resources`. I'm not saying that ordering makes more sense :sweat_smile: It just matches.

Reply:
Ahh good call. I moved things around a bit because I think it's weird to split the `--record-X` options like that: [4a3af79](https://github.com/gazebosim/gz-sim/pull/1547/commits/4a3af79e77413881d894a5ef900a013cc14e9ed7)